File: 1.Basics/4. For Loops/solution.py
import math

def check_prime(n):
    if(n<2):
        return "No"
    for i in range(2,math.floor(math.sqrt(n)+1)):
        if(n%i == 0):
            return "No"
    return "Yes"


if __name__ == '__main__': 
    t = int(input ())
    for _ in range (t):
        num = int(input())
        print(check_prime(num))

File: 1.Basics/4. For Loops/solution.js
let readline = require("readline");

let count = 0;
let testcasesCount = null;

const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
});

const checkPrimeBruteForce = function (num) {
  if (num < 2) return "No";
  for (i = 2; i < n; i++) {
    if (num % i === 0) {
      return "No";
    }
  }
  return "Yes";
};

const checkPrime = function (num) {
  if (num < 2) return "No";
  for (i = 2; i <= Math.sqrt(num); i++) {
    if (num % i === 0) {
      return "No";
    }
  }
  return "Yes";
};

rl.on("line", (data) => {
  if (testcasesCount === null) {
    testcasesCount = parseInt(data);
  } else {
    count++;
    console.log(checkPrime(parseInt(data)));
    if (count === testcasesCount) {
      rl.close();
    }
  }
});
